flow/connectors/postgres/qrep.go

```diff
@@ -295,6 +302,91 @@ func (c *PostgresConnector) getNumRowsPartitions(
 	}
 }
 
+func (c *PostgresConnector) getCTIDBlockPartitions(
+	ctx context.Context,
+	tx pgx.Tx,
+	parsedWatermarkTable utils.SchemaTable,
+	numPartitions int64,
+	last *protos.QRepPartition,
+) ([]*protos.QRepPartition, error) {
+	// Compute total heap blocks for the table using relation size and block_size.
+	blocksQuery := "SELECT (pg_relation_size(to_regclass($1)) / current_setting('block_size')::int)::bigint"
+	var totalBlocks pgtype.Int8
+	if err := tx.QueryRow(ctx, blocksQuery, parsedWatermarkTable.String()).Scan(&totalBlocks); err != nil {
+		return nil, fmt.Errorf("failed to get relation blocks: %w", err)
+	}
+	if totalBlocks.Int64 <= 0 {
+		return nil, nil
+	}
+
+	type tid struct {
+		b uint32
+		o uint32
+	}
+	cmp := func(a tid, b tid) int {
```

Comment:
should mirror https://github.com/PeerDB-io/peerdb/blob/c020dd474098a73aeb2c093eb505c2666b0dcc45/flow/connectors/utils/partition.go#L224 using `cmp.Compare`